gitlab/resource_gitlab_project_approval_rule.go

```diff
@@ -0,0 +1,254 @@
+package gitlab
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+// https://docs.gitlab.com/ee/api/merge_request_approvals.html#create-project-level-rule
+
+func resourceGitlabProjectApprovalRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabProjectApprovalRuleCreate,
+		Read:   resourceGitlabProjectApprovalRuleRead,
+		Update: resourceGitlabProjectApprovalRuleUpdate,
+		Delete: resourceGitlabProjectApprovalRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"approvals_required": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"user_ids": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeInt},
+				Set:      schema.HashInt,
+			},
+			"group_ids": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeInt},
+				Set:      schema.HashInt,
+			},
+			"eligible_approvers": {
```

Comment:
Is there a use case for having this attribute? The reason I am asking is that the actual GitLab API returns a list of objects with some more fields (`name`, `username`, etc.) If we include `eligible_approvers` it would be better if it [matched the upstream API (see design principle)](https://www.terraform.io/docs/extend/hashicorp-provider-design-principles.html#resource-and-attribute-schema-should-closely-match-the-underlying-api). We can also decide to omit this attribute in this PR and add it later if somebody needs it.

Reply:
I decided to remove `eligible_approvers`. I think we can add it later, if needed. Thanks!